java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -331,10 +331,16 @@ abstract class AbstractArcHost(
             return
         }
 
-        // Instantiate each particle and its handles in a ParticleContext.
-        for (particleSpec in partition.particles) {
-            val particleContext = setUpParticleAndHandles(particleSpec, context)
-            context.particles[particleSpec.particleName] = particleContext
+        for (idx in 0 until partition.particles.size) {
```

Comment:
partition.particles.forEachIndexed is more idiomatic

Reply:
there's a break which makes forEach/forEachIndexed awkward.

Reply:
it's an nit, so not required. the code is functionally the same. Just don't try to use this PR to get Kotlin readability :)